File: app/src/main/java/com/worldventures/dreamtrips/wallet/analytics/oncard/action/SmartCardSystemAction.java
package com.worldventures.dreamtrips.wallet.analytics.oncard.action;

import com.worldventures.dreamtrips.core.utils.tracksystem.AdobeTracker;
import com.worldventures.dreamtrips.core.utils.tracksystem.AnalyticsEvent;

import io.techery.janet.smartcard.model.analytics.AnalyticsLog;
import io.techery.janet.smartcard.model.analytics.AnalyticsLogEntryRestart;

@AnalyticsEvent(action = "wallet:oncard:system", trackers = AdobeTracker.TRACKER_KEY)
class SmartCardSystemAction extends SmartCardAnalyticsAction {

   SmartCardSystemAction(AnalyticsLog logEntry) {
      super(logEntry);
   }

   @Override
   protected void processLog(int type, AnalyticsLog logEntry) {
      super.processLog(type, logEntry);
      switch (type) {
         case AnalyticsLog.RESTART:
            AnalyticsLogEntryRestart restart = (AnalyticsLogEntryRestart) logEntry;
            attributeMap.put("ocrestart", "1");
            attributeMap.put("ocrestartcode", String.valueOf(restart.reason()));
            break;
         case AnalyticsLog.SET_TIME:
            attributeMap.put("ocsettime", "1");
            break;
      }
   }

}

File: wallet/src/main/java/com/worldventures/wallet/analytics/oncard/action/SmartCardSystemAction.kt
package com.worldventures.wallet.analytics.oncard.action

import com.worldventures.janet.analytics.AnalyticsEvent
import com.worldventures.core.service.analytics.AdobeTracker

import io.techery.janet.smartcard.model.analytics.AnalyticsLog
import io.techery.janet.smartcard.model.analytics.AnalyticsLogEntryRestart

@AnalyticsEvent(action = "wallet:oncard:system", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
internal class SmartCardSystemAction(logEntry: AnalyticsLog) : SmartCardAnalyticsAction(logEntry) {

   override fun processLog(type: Int, logEntry: AnalyticsLog) {
      super.processLog(type, logEntry)
      when (type) {
         AnalyticsLog.RESTART -> {
            val restart = logEntry as AnalyticsLogEntryRestart
            attributeMap.put("ocrestart", "1")
            attributeMap.put("ocrestartcode", restart.reason().toString())
         }
         AnalyticsLog.SET_TIME -> attributeMap.put("ocsettime", "1")
      }
   }

}
